src/backend/common/deferred/__init__.py

```diff
@@ -38,6 +43,124 @@ def defer(
     queue.enqueue(task, url=_url, headers=headers, service=service)
 
 
+def enqueue(
+    *,
+    url: str,
+    headers: Dict[str, str] = {},
+    method: Optional[str] = "POST",
+    params: Optional[Dict[Any, Any]] = None,
+    queue_name: Optional[str] = _DEFAULT_QUEUE,
+    target: Optional[str] = None
+):
+    # TODO: Add support for `name`, `countdown`, `eta`, `retry_options`
+    defer(_run_enqueue, url, method, params, _target=target, _headers=headers, _queue=queue_name)
+
+
+def _run_enqueue(
+    url: str,
+    method: str,
+    params: Optional[Dict[Any, Any]] = None
+):
+    o = urlparse(url)
+
+    # Do not allow hitting URLs outside of TBA services via enqueue
+    if o.netloc:
+        logging.error(
+            'Detected an attempt to reach an outside server via enqueued task.'
+        )
+
+    netloc = request.headers.get('X-Appengine-Default-Version-Hostname')
+    if not netloc:
+        abort(403)
+
+    headers = {}
+
+    if method == "POST":
+        if params:
+            headers["Content-Type"] = "application/x-www-form-urlencoded"
+    elif method == "GET":
+        query = parse_qs(o.query)
+        print(query)
+        # query.update(params) # TODO: Some sort of
+        print(urlencode(_flatten_params(params)))
+        # TODO: Add the other things here...
+    else:
+        raise Exception("Unsupported enqueue method - only `GET` and `POST` supported.")
+
+    # if params:
+    #     # TODO: url-encode this shit - see the code below
+    #     pass
+
+    o = o._replace(scheme=request.scheme, netloc=netloc)
+
+    #   if payload and params:
+    #     raise InvalidTaskError('Message body and parameters both present for '
+    #                            'POST method; only one of these can be '
+    #                            'supplied')
+    #   elif query:
+    #     raise InvalidTaskError('POST method cannot have a query string; '
+    #                            'use the "params" keyword argument instead')
+    #   elif params:
+    #     self.__payload = Task.__encode_params(params)
+    #     self.__headers.setdefault(
+    #         'content-type', 'application/x-www-form-urlencoded')
+    #   elif payload is not None:
+    #     self.__payload = Task.__convert_payload(payload, self.__headers)
+
+    url = o.geturl()
+
+    # Add an auth header to support service-to-service requests
+    token = id_token.fetch_id_token(auth_requests.Request(), url)
+    headers = {
+        'Authorization': f'Bearer {token}'
+    }
+
+    import requests as reqs
+
+    # if params:
+    #     # URL encode params, set
+    #     # application/x-www-form-urlencoded
+    #     return urllib.urlencode(_flatten_params(params))
+
+    # Non-200 status codes should be retried, yeah?
+    # Should not retry 401 or 404
```

Comment:
- [ ] Do this. We need to look at the status code - or maybe we can have requests throw on non-200 status codes or something.